### PR TITLE
Fix embedding normalization for lookup embedder.

### PR DESCRIPTION
**Pull request by samuelbroscheit**
Was there because of TransE. Now using the functionality provided by Pytorch and discarding the normalization with grad.

### CODE REVIEW

**Comment by rgemulla** [comment]
Please make a new PR off master with just the changes to the embedder.

File: kge/model/embedder/lookup_embedder.yaml [hunk 1]
```diff
@@ -39,9 +39,6 @@ lookup_embedder:
     # l_p norm to use. Negative numbers mean do not normalize.
     p: -1.                    # common choices: 1., 2.
 
-    # Whether to push gradients through normalization.
-    with_grad: False
-
```
Comment (rgemulla) [inline kge/model/embedder/lookup_embedder.yaml:44]: Good to remove this, but make sure the old configs still run (when this is set to false).

File: kge/model/embedder/lookup_embedder.py [hunk 2]
```diff
@@ -29,9 +28,18 @@ def __init__(
...
-            self.vocab_size, self.dim, sparse=self.sparse
-        )
+        if self.normalize_p > 0:
+            self._embeddings = torch.nn.Embedding(
+                self.vocab_size,
+                self.dim,
+                sparse=self.sparse,
+                norm_type=self.normalize_p,
+                max_norm=1.0,
+            )
```
Comment (rgemulla) [inline kge/model/embedder/lookup_embedder.py:38]: This is not correct. We also want vectors of norm < 1 to be normalized.

Comment (samuelbroscheit) [reply kge/model/embedder/lookup_embedder.py:38]: Yes I wanted to discuss this properly this time: The only code example for embedding normalization in TransE that I found was doing it like this, i.e. only normalize when norm > 1. We were not sure why and ad hoc decided to always normalize. So the question is now what we want to do: do we replicate what somebody else did for TransE or just always normalize.

File: kge/model/embedder/lookup_embedder.yaml (hunk 1 above)
Comment (samuelbroscheit) [reply kge/model/embedder/lookup_embedder.yaml:44]: So throw error when "with_grad: True" and run for "with_grad: False"?

Comment (rgemulla) [reply kge/model/embedder/lookup_embedder.yaml:44]: Yes, at the place where we process deprecated options.

File: kge/model/embedder/lookup_embedder.py (hunk 2 above)
Comment (rgemulla) [reply kge/model/embedder/lookup_embedder.py:38]: The TransE paper proposes to normalize. If we wanted max_norm (we do not for now, I think), it should be a separate option.

File: kge/model/embedder/lookup_embedder.yaml (hunk 1 above)
Comment (samuelbroscheit) [reply kge/model/embedder/lookup_embedder.yaml:44]: I know if you look at the changed files this partly already done in this PR.

Comment (rgemulla) [reply kge/model/embedder/lookup_embedder.yaml:44]: Yes, once you make this PR off master it will be easy to find ;)

File: kge/model/embedder/lookup_embedder.py (hunk 2 above)
Comment (samuelbroscheit) [reply kge/model/embedder/lookup_embedder.py:38]: From Bordes' implementation seems that he always normalizes as well:

https://github.com/glorotxa/SME/blob/a1aa9fdb790fd356a17cd13d2b7b75feba2d8c82/FB15k_exp.py#L267

https://github.com/glorotxa/SME/blob/a1aa9fdb790fd356a17cd13d2b7b75feba2d8c82/model.py#L277

(I vaguely remembering looking at this code already, maybe this why we chose to always normalize back then).
